libobs/util/platform-nix-libsecret.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023 Dennis Sädtler <dennis@obsproject.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include <gio/gio.h>
+#include <libsecret/secret.h>
+
+#include "platform.h"
+#include "bmem.h"
+
+static const SecretSchema obs_schema = {
+	"com.obsproject.Studio",
```

Comment:
Unfortunately a reference to `com.obsproject.Studio` has nothing to do inside libobs, it can only be inside OBS Studio which is the UI.

Reply:
We refer to libobs as the "OBS Studio Library" so I think it's fair to use it here as well. Because that's what it is.
https://github.com/obsproject/obs-studio/blob/master/libobs/pkgconfig/libobs.pc.in#L7

Reply:
`com.obsproject.Studio` represent an application not a library.

Reply:
The name here is indeed incorrect, as it is supposed to identify a unique schema. I'll change it to something like `com.obsproject.Studio.Secret` I don't see any point in not using that here. The distinction between libobs and OBS Studio is a technicality that just doesn't matter since libobs should never be packaged or shipped as a separate library (I know some do, but that's irrelevant to us).

Reply:
Schema name aside, currently entries are prefixed or identified as being from "OBS Studio" as well, if we wanted to avoid that, the API user would need to be able to specify the name so the user-visible entries in the keychain make sense.

For that purpose the API could be amended with another parameter, e.g. `os_keychain_save(const char* namespace, const char *key, const char *data)` so that the UI could then specify `os_keychain_save("OBS Studio", ...)`.